pkg/funcs/functions.go

```diff
@@ -0,0 +1,119 @@
+package pkg
+
+import (
+	"time"
+
+	neo "github.com/cloudprivacylabs/lsa-neo4j"
+	"github.com/cloudprivacylabs/lsa/pkg/types"
+	oc "github.com/cloudprivacylabs/opencypher"
+	"github.com/neo4j/neo4j-go-driver/neo4j"
+)
+
+func init() {
+	oc.RegisterGlobalFunc(oc.Function{
+		Name:      "age",
+		MinArgs:   1,
+		MaxArgs:   2,
+		ValueFunc: AgeFunc,
+	})
+}
+
+func AgeFunc(ctx *oc.EvalContext, args []oc.Value) (oc.Value, error) {
+	if args[0].Get() == nil {
+		return oc.RValue{}, nil
+	}
+	var n4jDate neo4j.Date
+	var n4jDateTime neo4j.LocalDateTime
+	var err error
+	// check if argument a neo4j.Date or neo4j.LocalDateTime
```

Comment:
Create a time.Time{} at the top, with each if-statement, parse it using one of the neo4j types

goDate:=time.Time{}
if dt, ok:=args[0].Get().(neo4j.DateTime); ok {
    goDate=dt.Time()
}
...


now:=time.Now()
// Same if blocks with the second arg